### PR TITLE
Bump `node-sass` to a version that uses a newer `libsass`

### LINKED ISSUE #1071: CVE-2018-19839 (Medium) detected in node-sass-8.0.0.tgz
## CVE-2018-19839 - Medium Severity Vulnerability
<details><summary> Vulnerable Library - <b>node-sass-8.0.0.tgz</b></p></summary>

<p>Wrapper around libsass</p>
<p>Library home page: <a href="https://registry.npmjs.org/node-sass/-/node-sass-8.0.0.tgz">https://registry.npmjs.org/node-sass/-/node-sass-8.0.0.tgz</a></p>
<p>

Dependency Hierarchy:
  - @osd/ui-framework-1.0.0.tgz (Root Library)
    - :x: **node-sass-8.0.0.tgz** (Vulnerable Library)
<p>Found in HEAD commit: <a href="https://github.com/opensearch-project/OpenSearch-Dashboards/commit/4fd064970b66ce555f48c22dfab6ed965d0e260a">4fd064970b66ce555f48c22dfab6ed965d0e260a</a></p>
<p>Found in base branch: <b>main</b></p>
</p>
</details>
<p></p>
<details><summary> Vulnerability Details</summary>
<p>  
  
In LibSass prior to 3.5.5, the function handle_error in sass_context.cpp allows attackers to cause a denial-of-service resulting from a heap-based buffer over-read via a crafted sass file.

<p>Publish Date: 2018-12-04
<p>URL: <a href=https://www.mend.io/vulnerability-database/CVE-2018-19839>CVE-2018-19839</a></p>
</p>
</details>
<p></p>
<details><summary> CVSS 3 Score Details (<b>6.5</b>)</summary>
<p>

Base Score Metrics:
- Exploitability Metrics:
  - Attack Vector: Network
  - Attack Complexity: Low
  - Privileges Required: None
  - User Interaction: Required
  - Scope: Unchanged
- Impact Metrics:
  - Confidentiality Impact: None
  - Integrity Impact: None
  - Availability Impact: High
</p>
For more information on CVSS3 Scores, click <a href="https://www.first.org/cvss/calculator/3.0">here</a>.
</p>
</details>
<p></p>
<details><summary> Suggested Fix</summary>
<p>

<p>Type: Change files</p>
<p>Release Date: 2018-11-28</p>
<p>Fix Resolution: Replace or update the following files: checked.h, core.h</p>

</p>
</details>
<p></p>

### LINKED ISSUE #1074: CVE-2018-11694 (High) detected in node-sass-8.0.0.tgz
## CVE-2018-11694 - High Severity Vulnerability
<details><summary> Vulnerable Library - <b>node-sass-8.0.0.tgz</b></p></summary>

<p>Wrapper around libsass</p>
<p>Library home page: <a href="https://registry.npmjs.org/node-sass/-/node-sass-8.0.0.tgz">https://registry.npmjs.org/node-sass/-/node-sass-8.0.0.tgz</a></p>
<p>

Dependency Hierarchy:
  - @osd/ui-framework-1.0.0.tgz (Root Library)
    - :x: **node-sass-8.0.0.tgz** (Vulnerable Library)
<p>Found in HEAD commit: <a href="https://github.com/opensearch-project/OpenSearch-Dashboards/commit/4fd064970b66ce555f48c22dfab6ed965d0e260a">4fd064970b66ce555f48c22dfab6ed965d0e260a</a></p>
<p>Found in base branch: <b>main</b></p>
</p>
</details>
<p></p>
<details><summary> Vulnerability Details</summary>
<p>  
  
An issue was discovered in LibSass through 3.5.4. A NULL pointer dereference was found in the function Sass::Functions::selector_append which could be leveraged by an attacker to cause a denial of service (application crash) or possibly have unspecified other impact.

<p>Publish Date: 2018-06-04
<p>URL: <a href=https://www.mend.io/vulnerability-database/CVE-2018-11694>CVE-2018-11694</a></p>
</p>
</details>
<p></p>
<details><summary> CVSS 3 Score Details (<b>8.8</b>)</summary>
<p>

Base Score Metrics:
- Exploitability Metrics:
  - Attack Vector: Network
  - Attack Complexity: Low
  - Privileges Required: None
  - User Interaction: Required
  - Scope: Unchanged
- Impact Metrics:
  - Confidentiality Impact: High
  - Integrity Impact: High
  - Availability Impact: High
</p>
For more information on CVSS3 Scores, click <a href="https://www.first.org/cvss/calculator/3.0">here</a>.
</p>
</details>
<p></p>

### LINKED ISSUE #1080: CVE-2018-19797 (Medium) detected in node-sass-8.0.0.tgz
## CVE-2018-19797 - Medium Severity Vulnerability
<details><summary> Vulnerable Library - <b>node-sass-8.0.0.tgz</b></p></summary>

<p>Wrapper around libsass</p>
<p>Library home page: <a href="https://registry.npmjs.org/node-sass/-/node-sass-8.0.0.tgz">https://registry.npmjs.org/node-sass/-/node-sass-8.0.0.tgz</a></p>
<p>

Dependency Hierarchy:
  - @osd/ui-framework-1.0.0.tgz (Root Library)
    - :x: **node-sass-8.0.0.tgz** (Vulnerable Library)
<p>Found in HEAD commit: <a href="https://github.com/opensearch-project/OpenSearch-Dashboards/commit/4fd064970b66ce555f48c22dfab6ed965d0e260a">4fd064970b66ce555f48c22dfab6ed965d0e260a</a></p>
<p>Found in base branch: <b>main</b></p>
</p>
</details>
<p></p>
<details><summary> Vulnerability Details</summary>
<p>  
  
In LibSass 3.5.5, a NULL Pointer Dereference in the function Sass::Selector_List::populate_extends in SharedPtr.hpp (used by ast.cpp and ast_selectors.cpp) may cause a Denial of Service (application crash) via a crafted sass input file.

<p>Publish Date: 2018-12-03
<p>URL: <a href=https://www.mend.io/vulnerability-database/CVE-2018-19797>CVE-2018-19797</a></p>
</p>
</details>
<p></p>
<details><summary> CVSS 3 Score Details (<b>6.5</b>)</summary>
<p>

Base Score Metrics:
- Exploitability Metrics:
  - Attack Vector: Network
  - Attack Complexity: Low
  - Privileges Required: None
  - User Interaction: Required
  - Scope: Unchanged
- Impact Metrics:
  - Confidentiality Impact: None
  - Integrity Impact: None
  - Availability Impact: High
</p>
For more information on CVSS3 Scores, click <a href="https://www.first.org/cvss/calculator/3.0">here</a>.
</p>
</details>
<p></p>

### LINKED ISSUE #1083: CVE-2019-6286 (Medium) detected in node-sass-8.0.0.tgz
## CVE-2019-6286 - Medium Severity Vulnerability
<details><summary> Vulnerable Library - <b>node-sass-8.0.0.tgz</b></p></summary>

<p>Wrapper around libsass</p>
<p>Library home page: <a href="https://registry.npmjs.org/node-sass/-/node-sass-8.0.0.tgz">https://registry.npmjs.org/node-sass/-/node-sass-8.0.0.tgz</a></p>
<p>

Dependency Hierarchy:
  - @osd/ui-framework-1.0.0.tgz (Root Library)
    - :x: **node-sass-8.0.0.tgz** (Vulnerable Library)
<p>Found in HEAD commit: <a href="https://github.com/opensearch-project/OpenSearch-Dashboards/commit/4fd064970b66ce555f48c22dfab6ed965d0e260a">4fd064970b66ce555f48c22dfab6ed965d0e260a</a></p>
<p>Found in base branch: <b>main</b></p>
</p>
</details>
<p></p>
<details><summary> Vulnerability Details</summary>
<p>  
  
In LibSass 3.5.5, a heap-based buffer over-read exists in Sass::Prelexer::skip_over_scopes in prelexer.hpp when called from Sass::Parser::parse_import(), a similar issue to CVE-2018-11693.

<p>Publish Date: 2019-01-14
<p>URL: <a href=https://www.mend.io/vulnerability-database/CVE-2019-6286>CVE-2019-6286</a></p>
</p>
</details>
<p></p>
<details><summary> CVSS 3 Score Details (<b>6.5</b>)</summary>
<p>

Base Score Metrics:
- Exploitability Metrics:
  - Attack Vector: Network
  - Attack Complexity: Low
  - Privileges Required: None
  - User Interaction: Required
  - Scope: Unchanged
- Impact Metrics:
  - Confidentiality Impact: None
  - Integrity Impact: None
  - Availability Impact: High
</p>
For more information on CVSS3 Scores, click <a href="https://www.first.org/cvss/calculator/3.0">here</a>.
</p>
</details>
<p></p>

### LINKED ISSUE #1085: CVE-2018-20821 (Medium) detected in node-sass-8.0.0.tgz
## CVE-2018-20821 - Medium Severity Vulnerability
<details><summary> Vulnerable Library - <b>node-sass-8.0.0.tgz</b></p></summary>

<p>Wrapper around libsass</p>
<p>Library home page: <a href="https://registry.npmjs.org/node-sass/-/node-sass-8.0.0.tgz">https://registry.npmjs.org/node-sass/-/node-sass-8.0.0.tgz</a></p>
<p>

Dependency Hierarchy:
  - @osd/ui-framework-1.0.0.tgz (Root Library)
    - :x: **node-sass-8.0.0.tgz** (Vulnerable Library)
<p>Found in HEAD commit: <a href="https://github.com/opensearch-project/OpenSearch-Dashboards/commit/4fd064970b66ce555f48c22dfab6ed965d0e260a">4fd064970b66ce555f48c22dfab6ed965d0e260a</a></p>
<p>Found in base branch: <b>main</b></p>
</p>
</details>
<p></p>
<details><summary> Vulnerability Details</summary>
<p>  
  
The parsing component in LibSass through 3.5.5 allows attackers to cause a denial-of-service (uncontrolled recursion in Sass::Parser::parse_css_variable_value in parser.cpp).

<p>Publish Date: 2019-04-23
<p>URL: <a href=https://www.mend.io/vulnerability-database/CVE-2018-20821>CVE-2018-20821</a></p>
</p>
</details>
<p></p>
<details><summary> CVSS 3 Score Details (<b>6.5</b>)</summary>
<p>

Base Score Metrics:
- Exploitability Metrics:
  - Attack Vector: Network
  - Attack Complexity: Low
  - Privileges Required: None
  - User Interaction: Required
  - Scope: Unchanged
- Impact Metrics:
  - Confidentiality Impact: None
  - Integrity Impact: None
  - Availability Impact: High
</p>
For more information on CVSS3 Scores, click <a href="https://www.first.org/cvss/calculator/3.0">here</a>.
</p>
</details>
<p></p>
<details><summary> Suggested Fix</summary>
<p>

<p>Type: Change files</p>
<p>Release Date: 2019-04-15</p>
<p>Fix Resolution: Replace or update the following files: util_string.cpp, util_string.hpp, parser.hpp, parser.cpp</p>

</p>
</details>
<p></p>

### LINKED ISSUE #1087: CVE-2019-6283 (Medium) detected in node-sass-8.0.0.tgz
## CVE-2019-6283 - Medium Severity Vulnerability
<details><summary> Vulnerable Library - <b>node-sass-8.0.0.tgz</b></p></summary>

<p>Wrapper around libsass</p>
<p>Library home page: <a href="https://registry.npmjs.org/node-sass/-/node-sass-8.0.0.tgz">https://registry.npmjs.org/node-sass/-/node-sass-8.0.0.tgz</a></p>
<p>

Dependency Hierarchy:
  - @osd/ui-framework-1.0.0.tgz (Root Library)
    - :x: **node-sass-8.0.0.tgz** (Vulnerable Library)
<p>Found in HEAD commit: <a href="https://github.com/opensearch-project/OpenSearch-Dashboards/commit/4fd064970b66ce555f48c22dfab6ed965d0e260a">4fd064970b66ce555f48c22dfab6ed965d0e260a</a></p>
<p>Found in base branch: <b>main</b></p>
</p>
</details>
<p></p>
<details><summary> Vulnerability Details</summary>
<p>  
  
In LibSass 3.5.5, a heap-based buffer over-read exists in Sass::Prelexer::parenthese_scope in prelexer.hpp.

<p>Publish Date: 2019-01-14
<p>URL: <a href=https://www.mend.io/vulnerability-database/CVE-2019-6283>CVE-2019-6283</a></p>
</p>
</details>
<p></p>
<details><summary> CVSS 3 Score Details (<b>6.5</b>)</summary>
<p>

Base Score Metrics:
- Exploitability Metrics:
  - Attack Vector: Network
  - Attack Complexity: Low
  - Privileges Required: None
  - User Interaction: Required
  - Scope: Unchanged
- Impact Metrics:
  - Confidentiality Impact: None
  - Integrity Impact: None
  - Availability Impact: High
</p>
For more information on CVSS3 Scores, click <a href="https://www.first.org/cvss/calculator/3.0">here</a>.
</p>
</details>
<p></p>

### LINKED ISSUE #1092: CVE-2018-20190 (Medium) detected in node-sass-8.0.0.tgz
## CVE-2018-20190 - Medium Severity Vulnerability
<details><summary> Vulnerable Library - <b>node-sass-8.0.0.tgz</b></p></summary>

<p>Wrapper around libsass</p>
<p>Library home page: <a href="https://registry.npmjs.org/node-sass/-/node-sass-8.0.0.tgz">https://registry.npmjs.org/node-sass/-/node-sass-8.0.0.tgz</a></p>
<p>

Dependency Hierarchy:
  - @osd/ui-framework-1.0.0.tgz (Root Library)
    - :x: **node-sass-8.0.0.tgz** (Vulnerable Library)
<p>Found in HEAD commit: <a href="https://github.com/opensearch-project/OpenSearch-Dashboards/commit/4fd064970b66ce555f48c22dfab6ed965d0e260a">4fd064970b66ce555f48c22dfab6ed965d0e260a</a></p>
<p>Found in base branch: <b>main</b></p>
</p>
</details>
<p></p>
<details><summary> Vulnerability Details</summary>
<p>  
  
In LibSass 3.5.5, a NULL Pointer Dereference in the function Sass::Eval::operator()(Sass::Supports_Operator*) in eval.cpp may cause a Denial of Service (application crash) via a crafted sass input file.

<p>Publish Date: 2018-12-17
<p>URL: <a href=https://www.mend.io/vulnerability-database/CVE-2018-20190>CVE-2018-20190</a></p>
</p>
</details>
<p></p>
<details><summary> CVSS 3 Score Details (<b>6.5</b>)</summary>
<p>

Base Score Metrics:
- Exploitability Metrics:
  - Attack Vector: Network
  - Attack Complexity: Low
  - Privileges Required: None
  - User Interaction: Required
  - Scope: Unchanged
- Impact Metrics:
  - Confidentiality Impact: None
  - Integrity Impact: None
  - Availability Impact: High
</p>
For more information on CVSS3 Scores, click <a href="https://www.first.org/cvss/calculator/3.0">here</a>.
</p>
</details>
<p></p>

### LINKED ISSUE #1093: CVE-2018-19827 (High) detected in node-sass-8.0.0.tgz
## CVE-2018-19827 - High Severity Vulnerability
<details><summary> Vulnerable Library - <b>node-sass-8.0.0.tgz</b></p></summary>

<p>Wrapper around libsass</p>
<p>Library home page: <a href="https://registry.npmjs.org/node-sass/-/node-sass-8.0.0.tgz">https://registry.npmjs.org/node-sass/-/node-sass-8.0.0.tgz</a></p>
<p>

Dependency Hierarchy:
  - @osd/ui-framework-1.0.0.tgz (Root Library)
    - :x: **node-sass-8.0.0.tgz** (Vulnerable Library)
<p>Found in HEAD commit: <a href="https://github.com/opensearch-project/OpenSearch-Dashboards/commit/4fd064970b66ce555f48c22dfab6ed965d0e260a">4fd064970b66ce555f48c22dfab6ed965d0e260a</a></p>
<p>Found in base branch: <b>main</b></p>
</p>
</details>
<p></p>
<details><summary> Vulnerability Details</summary>
<p>  
  
In LibSass 3.5.5, a use-after-free vulnerability exists in the SharedPtr class in SharedPtr.cpp (or SharedPtr.hpp) that may cause a denial of service (application crash) or possibly have unspecified other impact.

<p>Publish Date: 2018-12-03
<p>URL: <a href=https://www.mend.io/vulnerability-database/CVE-2018-19827>CVE-2018-19827</a></p>
</p>
</details>
<p></p>
<details><summary> CVSS 3 Score Details (<b>8.8</b>)</summary>
<p>

Base Score Metrics:
- Exploitability Metrics:
  - Attack Vector: Network
  - Attack Complexity: Low
  - Privileges Required: None
  - User Interaction: Required
  - Scope: Unchanged
- Impact Metrics:
  - Confidentiality Impact: High
  - Integrity Impact: High
  - Availability Impact: High
</p>
For more information on CVSS3 Scores, click <a href="https://www.first.org/cvss/calculator/3.0">here</a>.
</p>
</details>
<p></p>

### PR DESCRIPTION
**Pull request by AMoo-Miki**
### Description

The original `node-sass` is deprecated and is built on top of a vulnerable version of `libsass`. This change employs a fork of `node-sass` that is built on a newer version of `libsass`.

### Issues Resolved

Fixes #1071
Fixes #1074
Fixes #1080
Fixes #1083
Fixes #1085
Fixes #1087
Fixes #1092
Fixes #1093

### Check List

- [X] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [X] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by joshuarrrr** [comment]
@AMoo-Miki There are some failing unit tests, as well as snapshots that require updates.

File: packages/osd-optimizer/package.json [hunk 1]
```diff
@@ -33,7 +33,6 @@
     "pirates": "^4.0.1",
     "postcss": "^8.4.5",
     "rxjs": "^6.5.5",
-    "node-sass": "^8.0.0",
```
Comment (BSFishy) [inline packages/osd-optimizer/package.json:36]: Why is this moved into `devDependencies`? Was it just originally accidentally included in `dependencies`?

**Review by BSFishy — APPROVED** [review]

File: packages/osd-optimizer/package.json (hunk 1 above)
Comment (AMoo-Miki) [reply packages/osd-optimizer/package.json:36]: Correct

File: packages/osd-interpreter/package.json [hunk 2]
```diff
@@ -27,7 +27,7 @@
     "del": "^6.1.1",
     "getopts": "^2.2.5",
     "pegjs": "0.10.0",
-    "sass-loader": "^10.4.1",
+    "sass-loader": "npm:@amoo-miki/sass-loader@10.4.1-node-sass-9.0.0-libsass-3.6.5",
```
Comment (joshuarrrr) [inline packages/osd-interpreter/package.json:30]: what a catchy name 😉

**Review by joshuarrrr — APPROVED** [review]
I'm excited for this fix, but have two questions:

1. Validation - did you diff the generated CSS output before and after the change? We don't necessarily have testing in place that would catch mangled SASS compilation.
2. Can we open a tracking issue about removing these patches with the criteria for being able to move back to the real versions? I know in this case we'll probably just ditch the deps altogether, but I still think it will be helpful in case we forget or get delayed.

File: packages/osd-interpreter/package.json (hunk 2 above)
Comment (AMoo-Miki) [reply packages/osd-interpreter/package.json:30]: hahahahaha

**Pull request merged**